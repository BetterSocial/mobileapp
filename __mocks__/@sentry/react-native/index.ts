export const addBreadcrumb = jest.fn();
