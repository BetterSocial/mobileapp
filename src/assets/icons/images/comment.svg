<svg width="20" height="18" viewBox="0 0 20 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M0 12.065V3.01626C0 1.35042 1.34315 0 3 0H17C18.6569 0 20 1.35042 20 3.01626V12.065C20 13.7309 18.6569 15.0813 17 15.0813H12.7515L10.1463 17.3732C8.5298 18.7953 6 17.6411 6 15.4816V15.0813H3C1.34315 15.0813 0 13.7309 0 12.065ZM7.76118 14.1276C7.42328 13.4981 6.76132 13.0704 6 13.0704H3C2.44772 13.0704 2 12.6203 2 12.065V3.01626C2 2.46098 2.44772 2.01084 3 2.01084H17C17.5523 2.01084 18 2.46098 18 3.01626V12.065C18 12.6203 17.5523 13.0704 17 13.0704H12.7515C12.267 13.0704 11.799 13.2472 11.4345 13.568L8.82925 15.8599C8.50596 16.1443 8 15.9135 8 15.4816V15.0813C8 15.0194 7.99722 14.9581 7.99177 14.8976C7.96792 14.6329 7.89306 14.383 7.77716 14.158L7.76118 14.1276ZM5 6C5 5.44772 5.44772 5 6 5H14C14.5523 5 15 5.44772 15 6C15 6.55228 14.5523 7 14 7H6C5.44772 7 5 6.55228 5 6ZM6 8C5.44772 8 5 8.44771 5 9C5 9.55229 5.44772 10 6 10H11C11.5523 10 12 9.55229 12 9C12 8.44771 11.5523 8 11 8H6Z" fill="black"/>
</svg>
