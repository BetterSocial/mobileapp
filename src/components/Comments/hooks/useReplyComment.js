

const useReplyComment = () => {

    const isLast = (item, index, countComment) => (
      index === countComment - 1 && (item.children_counts.comment || 0) === 0
    );

    const isLastInParent = (index, countComment) => index === countComment - 1;

  return {
    isLast,
    isLastInParent
  }

}

export default useReplyComment