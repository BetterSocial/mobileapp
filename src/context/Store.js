import * as React from 'react';

import {channelReducer, channelState} from './reducers/channelReducer';
import {clientReducer, clientState} from './reducers/clientReducer';
import {discoveryReducer, discoveryState} from './reducers/discoveryReducer';
import {domainReducer, domainState} from './reducers/domainReducer';
import {feedsReducer, feedsState} from './reducers/FeedReducer';
import {followingReducer, followingState} from './reducers/followingReducer';
import {generalComponentReducer, generalComponentState} from './reducers/generalComponentReducer';
import {groupChatReducer, groupChatState} from './reducers/groupChat';
import {initialStartupReducer, initialStartupState} from './reducers/intialStartupReducer';
import {localCommunityReducer, localCommunityState} from './reducers/localComunityReducer';
import {myProfileFeedReducer, myProfileFeedState} from './reducers/myProfileFeedReducer';
import {myProfileReducer, myProfileState} from './reducers/myProfileReducer';
import {newsReducer, newsState} from './reducers/newsReducer';
import {otherProfileFeedReducer, otherProfileFeedState} from './reducers/OtherProfileFeedReducer';
import {topicsReducer, topicsState} from './reducers/topicsReducer';
import {unReadMessageReducer, unReadMessageState} from './reducers/unReadMessageReducer';
import {usersReducer, usersState} from './reducers/userReducer';
import {commentReducer, commentState} from './reducers/commentReducer';

/**
 *
 * @typedef {Object} MainContext
 * @property {import('./reducers/discoveryReducer').DiscoveryState} discovery
 */
const Store = ({children}) => {
  const rootReducer = {
    channel: React.useReducer(channelReducer, channelState),
    client: React.useReducer(clientReducer, clientState),
    discovery: React.useReducer(discoveryReducer, discoveryState),
    feeds: React.useReducer(feedsReducer, feedsState),
    following: React.useReducer(followingReducer, followingState),
    generalComponent: React.useReducer(generalComponentReducer, generalComponentState),
    groupChat: React.useReducer(groupChatReducer, groupChatState),
    localCommunity: React.useReducer(localCommunityReducer, localCommunityState),
    myProfileFeed: React.useReducer(myProfileFeedReducer, myProfileFeedState),
    otherProfileFeed: React.useReducer(otherProfileFeedReducer, otherProfileFeedState),
    news: React.useReducer(newsReducer, newsState),
    profile: React.useReducer(myProfileReducer, myProfileState),
    topics: React.useReducer(topicsReducer, topicsState),
    unReadMessage: React.useReducer(unReadMessageReducer, unReadMessageState),
    users: React.useReducer(usersReducer, usersState),
    domains: React.useReducer(domainReducer, domainState),
    initialStartup: React.useReducer(initialStartupReducer, initialStartupState),
    comments: React.useReducer(commentReducer, commentState)
  };
  return <Context.Provider value={rootReducer}>{children}</Context.Provider>;
};
export const Context = React.createContext(null);
export const useAppContext = () => React.useContext(Context);

export default Store;
