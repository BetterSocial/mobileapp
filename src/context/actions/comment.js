import {SET_COMMENT_LIST} from '../Types';

export const saveComment = (payload, dispatch) => {
  dispatch({
    type: SET_COMMENT_LIST,
    payload
  });
};
