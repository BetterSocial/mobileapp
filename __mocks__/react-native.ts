import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN