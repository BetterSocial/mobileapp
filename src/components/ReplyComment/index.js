import * as React from 'react';
import {
  Pressable,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import ArrowLeftIcon from '../../../assets/icons/arrow-left.svg';
import ButtonHightlight from '../ButtonHighlight';
import Comment from '../Comments/Comment';
import ConnectorWrapper from '../Comments/ConnectorWrapper';
import ReplyCommentItem from '../Comments/ReplyCommentItem';
import StringConstant from '../../utils/string/StringConstant';
import WriteComment from '../Comments/WriteComment';
import useReplyComment from './hooks/useReplyComment';
import useWriteComment from '../Comments/hooks/useWriteComment';
import {fonts} from '../../utils/fonts';
import useCommentAction from '../Comments/hooks/useCommentAction';
import {COLORS} from '../../utils/theme';

const ReplyCommentId = ({
  itemProp,
  indexFeed,
  level,
  updateParent,
  page,
  dataFeed,
  updateReply,
  itemParent,
  getComment
}) => {
  const navigation = useNavigation();
  const {
    setCommentHook,
    temporaryText,
    isLastInParentHook,
    setTextComment,
    newCommentList,
    item,
    showChildrenCommentView,
    updateFeed,
    scrollViewRef,
    createComment,
    getThisComment
  } = useReplyComment({
    itemProp,
    indexFeed,
    dataFeed,
    updateParent,
    updateReply,
    itemParent,
    page,
    getComment
  });
  const {handleUsernameReplyComment} = useWriteComment();
  const {showAlertDelete} = useCommentAction();
  React.useEffect(() => {
    if (setTextComment && typeof setTextComment === 'function') {
      setTextComment(temporaryText);
    }
  }, [temporaryText]);

  React.useEffect(() => {
    if (itemProp) {
      getThisComment();
    }
  }, [itemProp]);

  const updateVote = () => {
    if (getComment && typeof getComment === 'function') {
      getComment();
    }
  };

  const navigationGoBack = () => navigation.goBack();
  if (!item) return null;
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar translucent={false} />
      <View style={styles.header}>
        <TouchableOpacity testID="backButton" onPress={navigationGoBack} style={styles.backArrow}>
          <ArrowLeftIcon width={20} height={20} fill="#000" />
        </TouchableOpacity>
        <Text testID="usernameText" style={styles.headerText}>
          Reply to {handleUsernameReplyComment(itemProp)}
        </Text>
        <View style={styles.btn} />
      </View>
      <ScrollView ref={scrollViewRef} contentContainerStyle={styles.commentScrollView}>
        <View style={styles.containerComment}>
          <ReplyCommentItem
            feedId={dataFeed?.id}
            indexFeed={indexFeed}
            user={item.user}
            comment={item}
            time={item.created_at}
            photo={item.user?.data?.profile_pic_url}
            isLast={newCommentList.length <= 0}
            level={level}
            refreshComment={updateFeed}
            updateVoteParent={getComment}
          />
          {newCommentList.length > 0 &&
            newCommentList.map((itemReply, index) => (
              <React.Fragment key={index}>
                {itemReply.user ? (
                  <ContainerReply>
                    <ConnectorWrapper index={index}>
                      <Pressable
                        onLongPress={() => {
                          showAlertDelete(itemReply, false, () => getThisComment(true));
                        }}
                        style={styles.childCommentWrapper}>
                        <Comment
                          indexFeed={indexFeed}
                          showLeftConnector={false}
                          time={itemReply.created_at}
                          photo={itemReply.user.data && itemReply.user.data.profile_pic_url}
                          isLast={level >= 2}
                          key={`r${index}`}
                          user={itemReply.user}
                          comment={itemReply}
                          onPress={() => showChildrenCommentView(itemReply)}
                          level={parseInt(level, 10) + 1}
                          updateVote={updateVote}
                          onLongPress={() => {
                            showAlertDelete(itemReply, false, () => getThisComment(true));
                          }}
                        />
                        {itemReply.children_counts.comment > 0 && (
                          <>
                            <View style={styles.seeRepliesContainer(isLastInParentHook(index))}>
                              <View style={styles.connector} />
                              <ButtonHightlight onPress={() => showChildrenCommentView(itemReply)}>
                                <Text style={styles.seeRepliesText}>
                                  {StringConstant.postDetailPageSeeReplies(
                                    itemReply.children_counts.comment || 0
                                  )}
                                </Text>
                              </ButtonHightlight>
                            </View>
                          </>
                        )}
                      </Pressable>
                    </ConnectorWrapper>
                  </ContainerReply>
                ) : null}
              </React.Fragment>
            ))}
          {newCommentList.length > 0 ? <View style={styles.childLevelMainConnector} /> : null}
        </View>
      </ScrollView>
      <WriteComment
        postId={item?.activity_id}
        inReplyCommentView={true}
        showProfileConnector={newCommentList.length > 0}
        username={handleUsernameReplyComment(itemProp)}
        onChangeText={setCommentHook}
        onPress={(isAnonimity, anonimityData) => {
          createComment(isAnonimity, anonimityData);
        }}
        value={temporaryText}
      />
    </SafeAreaView>
  );
};
export const ContainerReply = ({children, isGrandchild = true, key}) => (
  <View
    key={key}
    style={[
      styles.containerReply,
      {borderColor: isGrandchild ? COLORS.transparent : COLORS.balance_gray}
    ]}>
    {children}
  </View>
);

export default React.memo(ReplyCommentId);

export const styles = StyleSheet.create({
  container: {
    height: 'auto',
    flex: 1,
    backgroundColor: COLORS.white
    // backgroundColor: 'blue',
  },
  containerComment: {
    marginTop: 8,
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    paddingLeft: 36,
    paddingRight: 23
  },
  header: {
    marginRight: -20,
    marginBottom: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  containerReply: {
    borderLeftWidth: 1,
    width: '100%'
    // flex: 1,
  },
  seeRepliesContainer: (isLast) => ({
    display: 'flex',
    flexDirection: 'row',
    paddingBottom: 14,
    borderLeftColor: isLast ? COLORS.transparent : COLORS.balance_gray
  }),
  seeRepliesText: {
    color: COLORS.signed_primary
  },
  btn: {
    paddingVertical: 8,
    paddingHorizontal: 20,
    borderRadius: 20
  },
  btnText: {
    color: COLORS.white
  },
  headerText: {
    fontFamily: fonts.inter[600],
    fontSize: 14,
    marginLeft: -24,
    color: COLORS.black,
    alignSelf: 'center'
  },
  image: {
    width: 48,
    height: 48
  },
  input: {
    backgroundColor: COLORS.concrete,
    flex: 1,
    color: COLORS.b,
    padding: 10,
    marginLeft: 20,
    borderRadius: 8
  },
  comment: {
    flexDirection: 'row',
    paddingRight: 20,
    position: 'absolute',
    bottom: 0
  },
  post: {
    fontFamily: fonts.inter[400],
    fontSize: 16,
    color: COLORS.mine_shaft,
    marginLeft: 28
  },
  mainLeftConnector: {
    height: '100%',
    width: 1,
    position: 'absolute',
    backgroundColor: COLORS.balance_gray,
    left: 46,
    zIndex: -100
  },
  connector: {
    width: 10,
    height: 5,
    borderLeftWidth: 1.5,
    borderBottomWidth: 1.5,
    // borderBottomLeftRadius: 1,
    borderLeftColor: COLORS.balance_gray,
    borderBottomColor: COLORS.balance_gray,
    marginRight: 4,
    marginLeft: -1,
    borderBottomLeftRadius: 15 / 2,
    marginTop: 0
    // marginTop: 1
  },
  childCommentWrapper: {
    borderLeftColor: COLORS.balance_gray,
    borderLeftWidth: 1,
    flex: 1
  },
  childLevelMainConnector: {
    flex: 1,
    borderLeftWidth: 1,
    borderLeftColor: COLORS.balance_gray,
    marginLeft: 24
  },
  backArrow: {
    padding: 10,
    paddingLeft: 24,
    paddingVertical: 18,
    alignSelf: 'center'
  },
  commentScrollView: {
    minHeight: '100%',
    paddingBottom: 83
  },
  childCommentWrapperLoading: {
    flex: 1
  }
});
