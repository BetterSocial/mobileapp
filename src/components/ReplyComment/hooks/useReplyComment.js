import React from 'react';
import Toast from 'react-native-simple-toast';
import moment from 'moment';
import {useNavigation} from '@react-navigation/core';
import StringConstant from '../../../utils/string/StringConstant';
import useUpdateComment from '../../Comments/hooks/useUpdateComment';
import {Context} from '../../../context';
import {createChildComment} from '../../../service/comment';
import {getFeedDetail} from '../../../service/post';
import {getCommentChild} from '../../../service/feeds';

const useReplyComment = ({
  itemProp,
  indexFeed,
  dataFeed,
  updateParent,
  updateReply,
  itemParent,
  page
}) => {
  const [temporaryText, setTemporaryText] = React.useState('');
  const [textComment, setTextComment] = React.useState('');
  const [newCommentList, setNewCommentList] = React.useState([]);
  const [item, setItem] = React.useState(itemProp);
  const navigation = useNavigation();
  // const dispatch = useDispatch
  const scrollViewRef = React.useRef(null);
  const {updateComment} = useUpdateComment();
  const [profile] = React.useContext(Context).profile;
  const [defaultData] = React.useState({
    data: {count_downvote: 0, count_upvote: 0, text: textComment},
    id: newCommentList?.length + 1,
    kind: 'comment',
    updated_at: moment(),
    children_counts: {comment: 0},
    latest_children: {},
    user: {
      data: {
        ...itemProp.user.data,
        profile_pic_url: profile?.myProfile?.profile_pic_path,
        username: profile.myProfile.username
      },
      id: itemProp.user.id
    }
  });

  const initTextComment = (text) => {
    setTextComment(text);
  };
  const setCommentHook = (text) => {
    setTemporaryText(text);
  };

  const handleFirstTextCommentHook = () => {
    setTextComment(temporaryText);
  };

  const updateReplyPost = (comment, itemParentProps, commentId) => {
    if (itemParentProps) {
      const updateMyComment = itemParentProps.latest_children.comment.map((dComment) => {
        if (dComment.id === commentId) {
          return {
            ...dComment,
            latest_children: {...dComment.latest_children, comment},
            children_counts: {comment: comment.length}
          };
        }
        return {...dComment};
      });
      const replaceComment = {
        ...itemParentProps,
        latest_children: {...itemParentProps.latest_children, comment: updateMyComment}
      };
      setItem(replaceComment);
      setNewCommentList(updateMyComment);
    }
  };

  const getThisCommentHook = () => {
    let comments = [];
    if (
      itemProp.latest_children &&
      itemProp.latest_children.comment &&
      Array.isArray(itemProp.latest_children.comment)
    ) {
      comments = itemProp.latest_children.comment.sort(
        (a, b) => moment(a.updated_at).unix() - moment(b.updated_at).unix()
      );
    }
    return comments;
  };

  const updateReplyPostHook = (comment, itemParentProps, commentId) => {
    if (itemParentProps) {
      const updateMyComment = itemParentProps.latest_children.comment.map((dComment) => {
        if (dComment.id === commentId) {
          return {
            ...dComment,
            latest_children: {...dComment.latest_children, comment},
            children_counts: {comment: comment.length}
          };
        }
        return {...dComment};
      });
      const replaceComment = {
        ...itemParentProps,
        latest_children: {...itemParentProps.latest_children, comment: updateMyComment}
      };
      return {replaceComment, updateMyComment};
    }
    return {replaceComment: itemParentProps, updateMyComment: itemParentProps};
  };

  const isLastInParentHook = (index) => index === (item.children_counts.comment || 0) - 1;

  const findCommentAndUpdateHook = (id, data) => {
    const newComment = newCommentList.map((comment) => {
      if (comment.id === id) {
        return {...comment, data: data.data};
      }
      return {...comment};
    });
    setNewCommentList(newComment);
    return newComment;
  };

  const updateVoteParentPostHook = (data, dataVote, comment) => {
    const updateMyComment = comment.latest_children.comment.map((dComment) => {
      if (dComment.id === dataVote.activity_id) {
        return {...dComment, data: data.data.data};
      }
      return {...dComment};
    });
    setNewCommentList(updateMyComment);
    return updateMyComment;
  };

  const updateVoteLatestChildrenParentHook = (response, dataVote, comment) => {
    if (comment) {
      const updateData = comment.latest_children.comment.map((dComment) => {
        if (dComment.id === dataVote.parent) {
          const mapChildren = dComment.latest_children.comment.map((child) => {
            if (child.id === dataVote.id) {
              return {...child, data: response.data};
            }
            return {...child};
          });
          return {
            ...dComment,
            latest_children: {...dComment.latest_children, comment: mapChildren}
          };
        }
        return {...dComment};
      });
      setNewCommentList(updateData);
      return updateData;
    }
    return [];
  };

  const showChildrenCommentView = async (itemReply) => {
    const itemParentProps = await {
      ...itemProp,
      latest_children: {...itemProp.latest_children, comment: newCommentList}
    };
    navigation.push('ReplyComment', {
      item: itemReply,
      level: 2,
      indexFeed,
      dataFeed,
      updateParent,
      itemParent: itemParentProps,
      updateReply: (comment, parentProps, id) => updateReplyPost(comment, parentProps, id),
      updateVote: (data, dataVote) => updateVoteParentPostHook(data, dataVote, itemParentProps),
      updateVoteLatestChildren: (data, dataVote) =>
        updateVoteLatestChildrenParentHook(data, dataVote, itemParentProps),
      getComment: getThisComment
    });
  };

  const updateFeed = async (isSort) => {
    try {
      const data = await getFeedDetail(item.activity_id);

      handleUpdateFeed(data, isSort);
    } catch (e) {
      if (__DEV__) {
        console.log(e);
      }
    }
  };
  const handleUpdateFeed = (data, isSort) => {
    if (data) {
      let oldData = data.data;
      if (isSort) {
        oldData = {
          ...oldData,
          latest_reactions: {...oldData.latest_reactions, comment: oldData.latest_reactions.comment}
        };
      }

      if (updateParent) {
        updateParent(oldData);
      }
    }
  };
  const createComment = async (isAnonimity, anonimityData) => {
    let sendPostNotif = false;
    if (page !== 'DetailDomainScreen') {
      sendPostNotif = true;
    }

    const commentWillBeAddedData = {
      ...defaultData,
      data: {...defaultData.data, text: textComment}
    };
    if (isAnonimity) {
      commentWillBeAddedData.user = {};
      commentWillBeAddedData.data.is_anonymous = true;
      commentWillBeAddedData.data.anon_user_info_emoji_name = anonimityData.emojiName;
      commentWillBeAddedData.data.anon_user_info_emoji_code = anonimityData.emojiCode;
      commentWillBeAddedData.data.anon_user_info_color_name = anonimityData.colorName;
      commentWillBeAddedData.data.anon_user_info_color_code = anonimityData.colorCode;
    }
    if (!isAnonimity) {
      commentWillBeAddedData.user.data = {
        profile_pic_url: profile?.myProfile?.profile_pic_path,
        username: profile?.myProfile?.username
      };
    }

    setTemporaryText('');
    setNewCommentList([...newCommentList, commentWillBeAddedData]);

    try {
      if (textComment.trim() !== '') {
        const data = await createChildComment(
          textComment,
          item.id,
          item.user.id,
          sendPostNotif,
          dataFeed?.actor?.id,
          dataFeed.id,
          dataFeed.message,
          isAnonimity,
          anonimityData
        );
        scrollViewRef.current.scrollToEnd();
        if (data.code === 200) {
          const newComment = [
            ...newCommentList,
            {
              ...defaultData,
              id: data.data.id,
              activity_id: data.data.activity_id,
              user: data.data.user,
              data: data.data.data,
              is_you: true
            }
          ];
          setNewCommentList(newComment);
          if (typeof updateReply === 'function') {
            updateReply(newComment, itemParent, item.id);
          }
          updateFeed(true);
          updateComment(item.activity_id);
        } else {
          Toast.show(StringConstant.generalCommentFailed, Toast.LONG);
        }
      } else {
        // Toast.show('Comments are not empty', Toast.LONG);
        // setLoadingCMD(false);
      }
    } catch (error) {
      Toast.show(StringConstant.generalCommentFailed, Toast.LONG);
    }
  };
  const getThisComment = async (isUpdate) => {
    if (itemProp.latest_children.comment && Array.isArray(itemProp.latest_children.comment)) {
      if (!isUpdate) {
        setNewCommentList(itemProp.latest_children?.comment);
      }
    }
    const response = await getCommentChild({activity_id: item?.id, feed_id: item.activity_id});
    setNewCommentList(response.data);
  };

  return {
    getThisCommentHook,
    updateReplyPostHook,
    setTemporaryText,
    setCommentHook,
    temporaryText,
    handleFirstTextCommentHook,
    textComment,
    isLastInParentHook,
    findCommentAndUpdateHook,
    updateVoteParentPostHook,
    updateVoteLatestChildrenParentHook,
    setTextComment,
    newCommentList,
    setNewCommentList,
    defaultData,
    setItem,
    item,
    initTextComment,
    updateReplyPost,
    showChildrenCommentView,
    updateFeed,
    handleUpdateFeed,
    scrollViewRef,
    createComment,
    getThisComment
  };
};
export default useReplyComment;
