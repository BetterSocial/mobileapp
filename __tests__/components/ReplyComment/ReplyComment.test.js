import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react-native';
import { View } from 'react-native';

import ReplyComment, {styles, ContainerReply} from '../../../src/components/ReplyComment'
import Store from '../../../src/context/Store';
import * as CommentService  from '../../../src/service/comment';
import  useReplyComment from '../../../src/components/ReplyComment/hooks/useReplyComment'
import * as replyFunc from '../../../src/components/ReplyComment/hooks/useReplyComment'

jest.mock('react-native/Libraries/Pressability/usePressability')
jest.mock('react-native/Libraries/Components/Pressable/Pressable')
const mockGoBack = jest.fn()
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({ goBack: mockGoBack, push: jest.fn() }),
  useRoute: () => ({
    params: {}
  }),
}));

jest.mock('../../../src/components/ReplyComment/hooks/useReplyComment', () => {
  const original = jest. requireActual('../../../src/components/ReplyComment/hooks/useReplyComment')
  return {
     __esModule: true,
     default: jest.fn(original.default)
  }
})


describe('Reply comment should run correctly', () => {

    const itemProp = {
        activity_id: "ca556c7a-8aaf-11ed-bb93-12946de4e4d9",
        children_counts: {},
        created_at: "2023-01-06T00:59:32.534991Z",
        data: {
            count_downvote: 0,
            count_upvote: 0,
            isNotSeen: true,
            text: "Hi",
        },
        id: "e78e2b3a-718a-48eb-bef5-b4ffd0701d4d",
        kind: "comment",
        latest_children: {},
        parent: "eb8aaa17-5853-4d1a-9b56-f22794dd1bae",
        updated_at: "2023-01-06T00:59:32.534991Z",
        user: {
            created_at: "2022-06-10T13:11:53.385703Z",
            data: {
                human_id: "I4K3M10FGR78EWQQDNQ2",
                profile_pic_url: "https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg",
                username: "Agita",
            },
            id: "c6c91b04-795c-404e-b012-ea28813a2006",
            updated_at: "2022-07-29T12:54:03.879150Z"
        },
        user_id: "c6c91b04-795c-404e-b012-ea28813a2006"
    }
    const indexFeed = 0
    const level = 1
    const page ='PostDetailPage'
    const itemParent = undefined
    afterEach(cleanup)
    it('should match snapshot', () => {
        const {toJSON} = render(<ReplyComment itemProp={itemProp} itemParent={itemParent} page={page} level={level} indexFeed={indexFeed} />, {wrapper: Store})
        expect(toJSON).toMatchSnapshot()
    })

    it('Back Button should run correctly', async () => {
        const {getByTestId} = render(<ReplyComment itemProp={itemProp} itemParent={itemParent} page={page} level={level} indexFeed={indexFeed} />, {wrapper: Store})
        fireEvent.press(getByTestId('backButton'))
        expect(mockGoBack).toHaveBeenCalled()

    })

    it('isLast style should run correctly', () => {
        expect(styles.seeRepliesContainer(true)).toEqual({
                display: 'flex',
                flexDirection: 'row',
                paddingBottom: 14,
                borderLeftColor:'transparent'
        })
    })

    it('ContainerReply component should run correctly', () => {
        const {toJSON} = render(<ContainerReply ><View testID='childreb' /> </ContainerReply>)
        expect(toJSON).toMatchSnapshot()
    })

    it('createChildComment should be called', () => {
        const {getByTestId} = render(<ReplyComment itemProp={itemProp} itemParent={itemParent} page={page} level={level} indexFeed={indexFeed} />, {wrapper: Store})
        fireEvent.press(getByTestId('iscommentenable'))
        expect(useReplyComment).toHaveBeenCalled()
     
    })
    it('getThisComment should run on firs load', async () => {
        const mock1 = jest.fn()
        jest.spyOn(replyFunc, 'default').mockImplementation(() => ({getThisCommentHook: mock1, temporaryText: 'halo'}))
        render(<ReplyComment itemProp={itemProp} itemParent={itemParent} page={page} level={level} indexFeed={indexFeed} />, {wrapper: Store})
        expect(mock1).toHaveBeenCalled()


    })

})
