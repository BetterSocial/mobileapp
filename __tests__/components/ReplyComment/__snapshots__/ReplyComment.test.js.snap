// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reply comment should run correctly ContainerReply component should run correctly 1`] = `[Function]`;

exports[`Reply comment should run correctly should match snapshot 1`] = `[Function]`;
